### PR TITLE
Add config file and support basic auth

### PR DESCRIPTION
**Pull request by ffilippopoulos**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,3 +66,72 @@ lhctl  completion > /etc/bash_completion.d/lhctl
...
+- name: context-1
+  url: https://longhorn.context.one
+  user: test
+  pass: test
+- name: context-2
+  url: https://longhorn.context.two
+default: context-1
+```
+
+The desire context can be specified via `--context` flag, otherwise the default
```
Comment (ribbybibby) [inline README.md]:
```suggestion
The desired context can be specified via `--context` flag, otherwise the default
```

File: README.md [hunk 2]
```diff
@@ -47,9 +47,13 @@ Available Commands:
...
 
 Flags:
-  -h, --help         help for lhctl
-  -t, --toggle       Help message for toggle
-      --url string   longhorn manager url (example: http://10.88.1.3/v1)
+      --config string    config file (default is $HOME/.lhctl.yaml)
+      --context string   config file context
+  -h, --help             help for lhctl
+      --pass string      password for http request
+  -t, --toggle           Help message for toggle
```
Comment (ribbybibby) [inline README.md]: Outside the scope of this PR but is this flag used? This seems like a placeholder message.

File: cmd/root.go [hunk 3]
```diff
@@ -83,26 +166,28 @@ func InitManagerClient() {
...
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		// Search config in home directory with name ".lhctl" (without extension).
+		viper.AddConfigPath(home)
+		viper.SetConfigName(".lhctl")
+	}
+	//viper.AutomaticEnv() // read in environment variables that match
```
Comment (ribbybibby) [inline cmd/root.go]: IMO, unused code should be removed before reaching master - not just commented out.

File: cmd/root.go [hunk 4]
```diff
@@ -49,9 +57,12 @@ func init() {
...
 	// Placeholder for config file
-	//rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.lhctl.yaml)")
+	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.lhctl.yaml)")
 	// root persistent flag `url` will be available everywhere in the package
-	rootCmd.PersistentFlags().StringVar(&url, "url", "", "longhorn manager url (example: http://10.88.1.3/v1)")
+	rootCmd.PersistentFlags().StringVar(&urlFlag, "url", "", "longhorn manager url (example: http://10.88.1.3/v1)")
+	rootCmd.PersistentFlags().StringVar(&userFlag, "user", "", "user for http request")
+	rootCmd.PersistentFlags().StringVar(&passFlag, "pass", "", "password for http request")
+	rootCmd.PersistentFlags().StringVar(&contextFlag, "context", "", "config file context")
 	//rootCmd.MarkFlagRequired("url")
```
Comment (ribbybibby) [inline cmd/root.go]: IMO, unused code should be removed before reaching master - not just commented out.

**Review by ribbybibby — COMMENTED** [review]
Looks good to me. In the absence of anything constructive to say, I've left a few nitpicks.

File: README.md (hunk 2 above)
Comment (ffilippopoulos) [reply README.md]: yup, you are correct that came with cobra, I shall remove

**Review by george-angel — APPROVED** [review]

**Pull request merged**